apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClientAwarenessInterceptor.kt

```diff
@@ -0,0 +1,20 @@
+package com.apollographql.apollo
+
+import okhttp3.Interceptor
+import okhttp3.Response
+
+/**
+ * An [Interceptor] to add [Client Awareness](https://www.apollographql.com/docs/studio/client-awareness/).
+ *
+ * Add this interceptor to your [okhttp3.OkHttpClient] instance when creating your [ApolloClient]
+ */
+class ApolloClientAwarenessInterceptor(val clientName: String, val clientVersion: String): Interceptor {
```

Comment:
If we have Context somehow, `Context.packageName` is the client name I guess. It is the same as Build config.APLICATION_ID.

We can then use the same packageName to retrieve the version from packageManager.

Got this from [here](https://stackoverflow.com/questions/4616095/how-to-get-the-build-version-number-of-your-android-application/6593822#6593822)

```
try {
    PackageInfo pInfo = context.getPackageManager().getPackageInfo(getPackageName(), 0);
    String version = pInfo.versionName;
} catch (PackageManager.NameNotFoundException e) {
    e.printStackTrace();
}
```

Reply:
Can we not put this via android-support module? Or something like this so that we access Context.

Or I had the idea to use AndroidX startup library to retrieve the Application context and store it statically. It's pretty safe because there is only 1 application Context.

Reply:
> Can we not put this via android-support module? 

That would require adding a new dependency in `build.gradle[.kts]`, not sure this is easier than adding an interceptor. I suspect it's a common usage to have an interceptor already setup for authentication so adding the ApolloClientAwerenessInterceptor should be a single line in that case.

Also the functionnality should be available for non-Android users as well, which would mean splitting it between `apollo-android-support` (for default Android values) and `apollo-runtime` (for the general case). All in all it felt like a lot of hassle for something that's easy to explain with a bit of documentation.